storage/azure/src/main/java/io/aiven/kafka/tieredstorage/storage/azure/AzureBlobStorage.java

```diff
@@ -69,6 +74,16 @@ public void configure(final Map<String, ?> configs) {
 
         metricsPolicy = new MetricCollector(config);
 
+        final ProxyConfig proxyConfig = config.proxyConfig();
+        if (proxyConfig != null) {
+            proxyOptions = new ProxyOptions(ProxyOptions.Type.SOCKS5,
```

Comment:
seems other types of proxies are supported. Should we consider passing the type as config (defaulting to SOCKS5, maybe as the only supported type) if the other parameters are similar/same?

Reply:
The HTTP proxy is another option, but there's no plan to add it right now. See the comment above: https://github.com/Aiven-Open/tiered-storage-for-apache-kafka/pull/510#discussion_r1523023702